webserver.py

```diff
@@ -41,61 +43,205 @@ def get_conn():
+            self.data['plots'] = {}
+        if 'unlockCount' in self.data:      # temp
+            del self.data['unlockCount']
+        for i in range(GAME_CONFIG['field_width']):
+            for j in range(GAME_CONFIG['field_height']):
+                if 'plot' + str(i) + '_' + str(j) in self.data:
+                    if self.data['plot' + str(i) + '_' + str(j)]['locked'] == 0:
+                        self.data['plots'][str(i) + '_' + str(j)]['seedType'] = self.data['plot' + str(i) + '_' + str(j)]['seedType']
+                        self.data['plots'][str(i) + '_' + str(j)]['sowTime'] = self.data['plot' + str(i) + '_' + str(j)]['sowTime']
+                    del self.data['plot' + str(i) + '_' + str(j)]
+        for i in range(2):
+            for j in range(2):
+                if self.data['plots'].get(str(i) + '_' + str(j)) is None:
+                    self.data['plots'][str(i) + '_' + str(j)] = {'seedType': 0, 'sowTime': 0}
+        return self
+
+    def check_cash(self):
+        total_seed_count = 0
+        total_crops_planted = 0
+        for seed_id in GAME_CONFIG['seeds']:
+            total_seed_count += self.data['seedCounts'][seed_id]
+        for i in range(GAME_CONFIG['field_width']):
+            for j in range(GAME_CONFIG['field_height']):
+                if str(i) + '_' + str(j) in self.data['plots']:
+                    if self.has_planted_crop(i, j):
+                        total_crops_planted += 1
+        if total_seed_count == 0 and total_crops_planted == 0:
+            if self.data['resources']['cash'] < RECIPE_CONFIG['recipes'][GAME_CONFIG['firstSeed']]['cashCost']:
+                self.data['resources']['cash'] = RECIPE_CONFIG['recipes'][GAME_CONFIG['firstSeed']]['cashCost']
+
+    def has_planted_crop(self, x, y):
+        if self.get_plot(x, y)['seedType'] != 0:
+            return True
+        else:
+            return False
+
+    def get_plot(self, x, y):
+        return self.data['plots'][str(x) + "_" + str(y)]
+
+    def check_password(self, password):
+        if password != self.data['password']:
+            raise Unauthorized("Invalid password")
+
+    def buy(self, recipe_id):
+        recipe_data = RECIPE_CONFIG['recipes'][recipe_id]
+        for resource in self.data['resources']:
+            if self.data['resources'][resource] < recipe_data[resource + 'Cost']:
+                message = "Not enough " + resource + " to buy a %s seed." % recipe_data['name']
+                return message
+            if self.data['seedCounts'][recipe_id] == GAME_CONFIG['max_seed_count']:
+                message = "Can't buy any more %s seeds." % recipe_data['name']
+                return message
+            self.data['resources'][resource] -= recipe_data[resource + 'Cost']
+        self.data['seedCounts'][recipe_id] += 1
+        message = "Bought a %s seed." % recipe_data['name']
+        return message
+
+    def sell(self, seed_id):
+        if self.data['seedCounts'][seed_id] <= 0:
```

Comment:
You should also be checking `if seed_id in self.data['seedCounts']` before this check.  If you make a `def get_seed_count(seed_id):` method, you won't need the extra check because `seed_id` not being in the data will yield a 0